src/abstractdataframe/abstractdataframe.jl

```diff
@@ -405,29 +413,94 @@ describe(df)
+    if !issubset(stats, allowed_fields) 
+        disallowed_fields = setdiff(stats, allowed_fields)
+        not_allowed = "Field(s) not allowed: $disallowed_fields. "
+        allowed = "Allowed fields are: $allowed_fields."
+        throw(ArgumentError(not_allowed * allowed)) 
     end
-end
 
-function StatsBase.describe(io::IO, X::AbstractVector{Union{T, Missing}}) where T
-    missingcount = count(ismissing, X)
-    pmissing = 100 * missingcount/length(X)
-    if pmissing != 100 && T <: Real
-        show(io, StatsBase.summarystats(collect(skipmissing(X))))
-    else
-        println(io, "Summary Stats:")
+    # Define 4 functions for getting summary statistics 
+    # use a dict because we dont know which measures the user wants
+    function get_stats(col::AbstractArray{<:Real})
+        sumstats = summarystats(col)
+        Dict(
+            :mean => sumstats.mean,
+            :std => Compat.std(col, mean = sumstats.mean),
+            :min => sumstats.min,
+            :q25 => sumstats.q25,
+            :median => sumstats.median,
+            :q75 => sumstats.q75,
+            :max => sumstats.max,
+            :nmissing => nothing,
+            :nunique => nothing,
+            :eltype => eltype(col)
+        )
+    end
+    
+    function get_stats(col::AbstractArray{<:Union{Real, Missing}})
+        nomissing = collect(skipmissing(col))
+        sumstats = summarystats(nomissing)
+        Dict(
+            :mean => sumstats.mean,
+            :std => Compat.std(nomissing, mean = sumstats.mean),
+            :min => sumstats.min,
+            :q25 => sumstats.q25,
+            :median => sumstats.median,
+            :q75 => sumstats.q75,
+            :max => sumstats.max,
+            :nmissing => count(ismissing, col),
+            :nunique => nothing,
+            :eltype => Missings.T(eltype(col))
+        )
+    end
+
+    function get_stats(col::AbstractArray{>:Missing})
+        Dict(
+            :mean => nothing,
+            :std => nothing,
+            :min => nothing,
+            :q25 => nothing,
+            :median => nothing,
+            :q75 => nothing,
+            :max => nothing,
+            :nmissing => count(ismissing, col),
+            :nunique => length(unique(col)),
+            :eltype => Missings.T(eltype(col))
+        )    
+    end 
+    
+    function get_stats(col)
+        Dict(
+            :mean => nothing,
+            :std => nothing,
+            :min => nothing,
+            :q25 => nothing,
+            :median => nothing,
+            :q75 => nothing,
+            :max => nothing,
+            :nmissing => nothing,
+            :nunique => length(unique(col)),
+            :eltype => eltype(col)
+        )   
+    end
+
+    # Put the summary stats into the return dataframe
+    data = DataFrame()
+    data[:variable] = names(df)
+
+    # An array of Dicts for summary statistics
+    columns = [get_stats(col) for col in df.columns] 
```

Comment:
Better use `columns(df)`, as on 0.7 `df.columns` doesn't work due to overriding field access. But then the variable `columns` needs to use another name.

Reply:
Fixed.